ARM-wvd-templates/nestedtemplates/managedDisks-customimagevm.json

```diff
@@ -1,387 +1,473 @@
-            "copy": {
-                "name": "rdsh-domain-join-loop",
-                "count": "[parameters('rdshNumberOfInstances')]"
-            },
-            "properties": {
-                "publisher": "Microsoft.Compute",
-                "type": "JsonADDomainExtension",
-                "typeHandlerVersion": "1.3",
-                "autoUpgradeMinorVersion": true,
-                "settings": {
-                    "name": "[variables('domain')]",
-                    "ouPath": "[parameters('ouPath')]",
-                    "user": "[parameters('administratorAccountUsername')]",
-                    "restart": "true",
-                    "options": "3"
-                },
-                "protectedSettings": {
-                    "password": "[parameters('administratorAccountPassword')]"
-                }
+          },
+          "imageReference": {
+            "id": "[parameters('rdshImageSourceId')]"
+          }
+        },
+        "networkProfile": {
+          "networkInterfaces": [
+            {
+              "id": "[resourceId('Microsoft.Network/networkInterfaces',concat(parameters('rdshPrefix'), add(copyindex(), parameters('vmInitialNumber')), '-nic'))]"
             }
+          ]
         },
-        {
-            "apiVersion": "2018-10-01",
-            "type": "Microsoft.Compute/virtualMachines/extensions",
-            "name": "[concat(parameters('rdshPrefix'), add(copyindex(), parameters('vmInitialNumber')), '/', 'dscextension')]",
-            "location": "[parameters('location')]",
-            "dependsOn": ["rdsh-domain-join-loop"],
-            "copy": {
-                "name": "rdsh-dsc-loop",
-                "count": "[parameters('rdshNumberOfInstances')]"
-            },
-            "properties": {
-                "publisher": "Microsoft.Powershell",
-                "type": "DSC",
-                "typeHandlerVersion": "2.73",
-                "autoUpgradeMinorVersion": true,
-                "settings": {
-                    "modulesUrl": "[parameters('artifactsLocation')]",
-                    "configurationFunction": "Configuration.ps1\\AddSessionHost",
-                    "properties": {
-                        "hostPoolName": "[parameters('hostpoolName')]",
-                        "registrationInfoToken": "[parameters('hostpoolToken')]"
-                    }
-                }
-            }
+        "diagnosticsProfile": {
+          "bootDiagnostics": {
+            "enabled": false
+          }
+        },
+        "licenseType": "Windows_Client"
+      },
+      "zones": "[if(equals(parameters('availabilityOption'), 'AvailabilityZone'), array(parameters('availabilityZone')), variables('emptyArray'))]"
+    },
+    {
+      "apiVersion": "2018-10-01",
+      "type": "Microsoft.Compute/virtualMachines/extensions",
+      "name": "[concat(parameters('rdshPrefix'), add(copyindex(), parameters('vmInitialNumber')), '/', 'Microsoft.PowerShell.DSC')]",

```

Comment:
I don't know what all of the implications are of renaming it for users that have existing VMs deployed with the current DSC extension name . That should probably be tested to see if it fails.

Reply:
This is Dhruma's changes.  I believe she already tested.  @dhrump please kindly confirm.

Reply:
It would also be good to update [/ARM-wvd-templates/DSC/README.md](https://github.com/Azure/RDS-Templates/blob/master/ARM-wvd-templates/DSC/README.md) with this new name as it currently references "dscextension".

Reply:
This template only gets called when creating new VMs and it does not get executed on existing VMs so existing VM scenario need not be tested.

Reply:
@Philip can you please help update the readme. If it goes in separate PR then i will udpate it

Reply:
@dhrump Updated readme, please kindly review.

Reply:
Readme looks good to me